docs/ibc/light-clients/overview.md

```diff
@@ -1,3 +1,45 @@
 <!--
 order: 1
--->
\ No newline at end of file
+-->
+
+# Overview 
+
+Learn how to build IBC lightclient modules and fulfill the interfaces required to integrate with core IBC. {synopsis}
+
+The following aims to provide a high level IBC lightclient module developer guide. The lifecycle of IBC lightclients are created and maintained by the core IBC `MsgServer` and the abstractions provided by the `02-client` submodule. A lightclient module developer is only required to implement the methods and fulfil the expectations of a defined set of interfaces. In ibc-go these interfaces reside within the `core/modules/exported` package. 
+
+Throughout this guide the `07-tendermint` lightclient module may be referred to as a reference example.
+
+## Concepts and vocabulary
+
+This guide serves to be implementation specific with respect to ibc-go. Readers are expected to familiarize themselves with the IBC protocol specifications.
+Please refer to the defintions outlined in [ICS-002 Client Semantics](https://github.com/cosmos/ibc/tree/main/spec/core/ics-002-client-semantics#Definitions).
+
+### ClientState 
+
+ClientState is a term used to define the data structure which encapsulates opaque lightclient state. This refers to internal data that accommodate the semantics of a lightclient. This may be any arbitrary data such as:
```

Comment:
> This refers to internal data that accommodate the semantics of a lightclient.

I find this phrasing a bit broad. What do you mean by "internal data that accommodate the semantics"?

Reply:
Yeah, I made this intentionally broad as it seemed pretty broad from the spec. But what I was referring to is the fields on the type which make up the constraints/rules listed below:

```
- Constraints used for client updates.
- Constraints used for misbehaviour detection.
- Constraints used for state verification.
- Constraints used for client upgrades.
```

Happy to change or add more info if you have any suggestions. I can try to think of how to rephrase it also.

Reply:
Awesome, thanks for the suggestion. I've gone with the following for now, we can always revisit this when we get some user feedback.

> ClientState is a term used to define the data structure which encapsulates opaque light client state. The `ClientState` contains all the information needed to verify a `ClientMessage` and perform membership and non-membership proof verification of counterparty state. This includes properties that refer to the remote state machine, the light client type and the specific light client instance.